### DISCUSSION TITLE
How can i get the available languages in a font?

### DISCUSSION THREAD

**Discussion by isaacfink** [opentypejs/opentype.js#598] [Q&A]
I have a project that relies heavily on custom fonts and I need to be able to tell if a font is in a certain language, how can I do that?

**Comment by Connum** [comment]
You can get the avaialble scripts in a font from the gsub table:
```js
font.tables.gsub.scripts.map(o => o.tag)
```
e.g. running this in the console of the demo page at https://opentype.js.org/
will result in the output
```js
['DFLT', 'cyrl', 'grek', 'latn']
```

**Comment by vinuel** [comment]
Checking the language support of a font is not trivial and nothing that opentype.je would do on it's own. You could collect all unicode encoded characters in the font and check it with a database of languages. I've made it [here](https://fontdrop.info/#/languages), and also wrote some lines about it in the About section.
